### PR TITLE
Issue 216 add storage permission schema

### PR DESCRIPTION
**Pull request by FrancoisJ**

### CODE REVIEW

File: antarest/login/model.py [hunk 1]
```diff
@@ -67,3 +79,33 @@ def __eq__(self, o: Any) -> bool:
             and (o.name == self.name)
             and (o.role == self.role)
         )
+
+
+class Group(DTO, Base):  # type: ignore
+    __tablename__ = "groups"
+
+    id = Column(Integer, Sequence("user_id_seq"), primary_key=True)
```
Comment (pl-buiquang) [inline antarest/login/model.py]: create a new sequence table "group_id_seq"

**Review by pl-buiquang — CHANGES_REQUESTED** [review]

File: antarest/login/model.py (hunk 1 above)
Comment (FrancoisJ) [reply antarest/login/model.py]: done thanks

File: antarest/login/web.py [hunk 2]
```diff
@@ -49,49 +49,81 @@ def auth() -> Any:
     def login() -> Any:
         return render_template("login.html")
 
-    @bp.route("/users", methods=["GET", "POST"], defaults={"id": None})
+    @bp.route("/users", methods=["GET"])
+    @jwt_required  # type: ignore
+    def users_get_all() -> Any:
+        if get_jwt_identity()["role"] != "ADMIN":
```
Comment (pl-buiquang) [inline antarest/login/web.py:55]: à remplacer par une méthode (isAdmin()) qui wrap ce code (comme ça on peut dire que "ADMIN" sera pas vraiment une magic string)

File: antarest/login/web.py [hunk 3]
```diff
@@ -49,49 +49,81 @@ def auth() -> Any:
     def login() -> Any:
         return render_template("login.html")
 
-    @bp.route("/users", methods=["GET", "POST"], defaults={"id": None})
+    @bp.route("/users", methods=["GET"])
+    @jwt_required  # type: ignore
```
Comment (pl-buiquang) [inline antarest/login/web.py:53]: Wrap the decorator inside another decorator to disable it when login is disabled in config. eg. https://stackoverflow.com/questions/5952641/decorating-decorators-try-to-get-my-head-around-understanding-it

**Review by pl-buiquang — CHANGES_REQUESTED** [review]
Protect storage module routes

File: antarest/login/web.py (hunk 3 above)
Comment (FrancoisJ) [reply antarest/login/web.py:53]: yes, i will do that in other issue, when we refactor storage

**Pull request merged**